strange error when loading config for new funloc dataset

locally when running `pytest -k session_specific_mri` (which is being added in #1039) I get the following traceback:

```
mne_bids_pipeline/tests/test_run.py:365: in test_session_specific_mri
    main()
mne_bids_pipeline/_main.py:221: in main
    config_imported = _import_config(
mne_bids_pipeline/_config_import.py:79: in _import_config
    config = SimpleNamespace(**{k: getattr(config, k) for k in keep_names})
E   AttributeError: 'types.SimpleNamespace' object has no attribute '_n_jobs'. Did you mean: 'n_jobs'?
```

it's the inner `getattr(config, k)` part that fails when `k` is `_n_jobs`. If I insert an error just before the problematic line, then try `pytest -k ds000247 --pdb` then `_n_jobs` isn't in the `config`... it's only added if `extra_config` is present:

https://github.com/mne-tools/mne-bids-pipeline/blob/8408697da094c04b0f05ff90a634545439aef6f7/mne_bids_pipeline/_config_import.py#L58-L66

Which I think means there's an implicit assumption that if `extra_config` is defined, that it *must* include a `_n_jobs` key.  Off the top of my head I don't see why that condition necessarily should hold, though prior to adding `funloc` it *did* incidentally hold: in `tests/test_run.py` `_n_jobs` shows up 3 times, (in the `extra_config` for ERP_CORE ERN, ds000248_base, and ds000248_ica).

AFAICT options are:

1. only insert that key into `keep_names` if it's actually present in the `extra_config`
2. put a comment near where I hit the error, saying "FYI you *must* have `_n_jobs` in there if you provide an `extra_config`"

WDYT?  Do you know/notice something else I haven't thought of?

Bonus context:
- If I hack in a `check=True` the config check (which happens before the line triggering the traceback) passes. There really isn't anything special about the new `tests/configs/config_funloc.py`

--- a/docs/source/dev.md.inc
+++ b/docs/source/dev.md.inc
@@ -32,11 +32,9 @@
 - Bundle all previously-external JavaScript to better preserve users' privacy (#982 by @hoechenberger)
 - Document the need for offscreen rendering support when running on headless servers (#997 by @drammock)
 
-### :medical_symbol: Code health
+### :medical_symbol: Code health and infrastructure
 
 - Switch from using relative to using absolute imports (#969 by @hoechenberger)
 - Enable strict type checking via mypy (#995, #1013, #1016 by @larsoner)
-
-### :medical_symbol: Code health and infrastructure
-
 - Improve logging messages in maxwell filtering steps. (#893 by @drammock)
+- Validate extra config params passed in during testing. (#1044 by @drammock)

--- a/ignore_words.txt
+++ b/ignore_words.txt
@@ -1,1 +1,2 @@
 master
+indx

--- a/mne_bids_pipeline/_config_import.py
+++ b/mne_bids_pipeline/_config_import.py
@@ -58,11 +58,13 @@ def _import_config(
     if extra_config:
         msg = f"With testing config: {extra_config}"
         logger.info(**gen_log_kwargs(message=msg, emoji="override"))
-        _update_config_from_path(
-            config=config,
-            config_path=extra_config,
+        extra_names = _update_config_from_path(
+            config=config, config_path=extra_config, include_private=True
         )
-        extra_exec_params_keys = ("_n_jobs",)
+        # Update valid_extra_names as needed if test configs in tests/test_run.py change
+        valid_extra_names = set(("_n_jobs", "_raw_split_size", "_epochs_split_size"))
+        assert set(extra_names) - valid_extra_names == set(), extra_names
+        extra_exec_params_keys = tuple(set(["_n_jobs"]) & set(extra_names))
     keep_names.extend(extra_exec_params_keys)
 
     # Check it
@@ -139,6 +141,7 @@ def _update_config_from_path(
     *,
     config: SimpleNamespace,
     config_path: PathLike,
+    include_private: bool = False,
 ) -> list[str]:
     user_names = list()
     config_path = pathlib.Path(config_path).expanduser().resolve(strict=True)
@@ -156,7 +159,7 @@ def _update_config_from_path(
         if not key.startswith("__"):
             # don't validate private vars, but do add to config
             # (e.g., so that our hidden _raw_split_size is included)
-            if not key.startswith("_"):
+            if include_private or not key.startswith("_"):
                 user_names.append(key)
             val = getattr(custom_cfg, key)
             logger.debug(f"Overwriting: {key} -> {val}")

